test/base/src/androidTest/kotlin/io/realm/test/shared/QueryTests.kt

```diff
@@ -242,7 +242,9 @@ class QueryTests {
             assertEquals(1, channel.receive().size)
 
             realm.writeBlocking {
-                objects(QuerySample::class).delete()
+                query<QuerySample>()
+                    .find()
+                    .delete()
```

Comment:
We are actually able to delete objects directly from the query. Don't know if there is a performance penalty of having to `find` the objects first. But should probably be investigated.

Reply:
https://github.com/realm/realm-core/blob/master/src/realm.h#L1842

Reply:
Hm, just checked it again and the C-API function isn't implemented yet. We even have an `ignore` exception in `realm.i` 😑